diff --git a/hw/block/Kconfig b/hw/block/Kconfig
index 737dbcd..3904e19 100644
--- a/hw/block/Kconfig
+++ b/hw/block/Kconfig
@@ -32,3 +32,6 @@ config VHOST_USER_BLK
 
 config SWIM
     bool
+
+config Q9_CF
+    bool
diff --git a/hw/block/meson.build b/hw/block/meson.build
index d646323..c4d6dcf 100644
--- a/hw/block/meson.build
+++ b/hw/block/meson.build
@@ -11,6 +11,7 @@ system_ss.add(when: 'CONFIG_PFLASH_CFI02', if_true: files('pflash_cfi02.c'))
 system_ss.add(when: 'CONFIG_SSI_M25P80', if_true: files('m25p80.c'))
 system_ss.add(when: 'CONFIG_SSI_M25P80', if_true: files('m25p80_sfdp.c'))
 system_ss.add(when: 'CONFIG_SWIM', if_true: files('swim.c'))
+system_ss.add(when: 'CONFIG_Q9_CF', if_true: files('q9_cf.c'))
 system_ss.add(when: 'CONFIG_XEN_BUS', if_true: files('xen-block.c'))
 
 system_ss.add(when: 'CONFIG_VIRTIO_BLK', if_true: files('virtio-blk.c'))
diff --git a/hw/char/Kconfig b/hw/char/Kconfig
index 020c0a8..49e5abf 100644
--- a/hw/char/Kconfig
+++ b/hw/char/Kconfig
@@ -95,3 +95,6 @@ config IP_OCTAL_232
     bool
     default y
     depends on IPACK
+
+config Q9_DUART68681
+    bool
diff --git a/hw/char/meson.build b/hw/char/meson.build
index fc3d7ee..8eafef7 100644
--- a/hw/char/meson.build
+++ b/hw/char/meson.build
@@ -25,6 +25,7 @@ system_ss.add(when: 'CONFIG_DIVA_GSP', if_true: files('diva-gsp.c'))
 
 system_ss.add(when: 'CONFIG_AVR_USART', if_true: files('avr_usart.c'))
 system_ss.add(when: 'CONFIG_COLDFIRE', if_true: files('mcf_uart.c'))
+system_ss.add(when: 'CONFIG_Q9_DUART68681', if_true: files('q9_duart68681.c'))
 system_ss.add(when: 'CONFIG_DIGIC', if_true: files('digic-uart.c'))
 system_ss.add(when: 'CONFIG_EXYNOS4', if_true: files('exynos4210_uart.c'))
 system_ss.add(when: 'CONFIG_MAX78000_UART', if_true: files('max78000_uart.c'))
diff --git a/hw/m68k/Kconfig b/hw/m68k/Kconfig
index aff769b..152f59d 100644
--- a/hw/m68k/Kconfig
+++ b/hw/m68k/Kconfig
@@ -46,3 +46,19 @@ config M68K_VIRT
     select GOLDFISH_TTY
     select GOLDFISH_RTC
     select VIRTIO_MMIO
+
+config Q9_TIMER_IRQ
+    bool
+
+config Q9_REMAP
+    bool
+
+config Q9BOARD
+    bool
+    default y
+    depends on M68K
+    select Q9_RTC72421
+    select Q9_TIMER_IRQ
+    select Q9_DUART68681
+    select Q9_CF
+    select Q9_REMAP
diff --git a/hw/m68k/meson.build b/hw/m68k/meson.build
index 84bc68f..65b7eaf 100644
--- a/hw/m68k/meson.build
+++ b/hw/m68k/meson.build
@@ -4,5 +4,8 @@ m68k_ss.add(when: 'CONFIG_MCF5208', if_true: files('mcf5208.c', 'mcf_intc.c'))
 m68k_ss.add(when: 'CONFIG_NEXTCUBE', if_true: files('next-kbd.c', 'next-cube.c'))
 m68k_ss.add(when: 'CONFIG_Q800', if_true: files('q800.c', 'q800-glue.c'))
 m68k_ss.add(when: 'CONFIG_M68K_VIRT', if_true: files('virt.c'))
+m68k_ss.add(when: 'CONFIG_Q9BOARD', if_true: files('q9board.c'))
+m68k_ss.add(when: 'CONFIG_Q9_TIMER_IRQ', if_true: files('q9_timer_irq.c'))
+m68k_ss.add(when: 'CONFIG_Q9_REMAP', if_true: files('q9_remap.c'))
 
 hw_arch += {'m68k': m68k_ss}
diff --git a/hw/rtc/Kconfig b/hw/rtc/Kconfig
index 315b0e4..289438f 100644
--- a/hw/rtc/Kconfig
+++ b/hw/rtc/Kconfig
@@ -31,3 +31,6 @@ config RS5C372_RTC
     bool
     depends on I2C
     default y if I2C_DEVICES
+
+config Q9_RTC72421
+    bool
diff --git a/hw/rtc/meson.build b/hw/rtc/meson.build
index 6c87864..6860194 100644
--- a/hw/rtc/meson.build
+++ b/hw/rtc/meson.build
@@ -14,3 +14,4 @@ system_ss.add(when: 'CONFIG_LS7A_RTC', if_true: files('ls7a_rtc.c'))
 system_ss.add(when: 'CONFIG_ALLWINNER_H3', if_true: files('allwinner-rtc.c'))
 system_ss.add(when: 'CONFIG_MC146818RTC', if_true: files('mc146818rtc.c'))
 system_ss.add(when: 'CONFIG_RS5C372_RTC', if_true: files('rs5c372.c'))
+system_ss.add(when: 'CONFIG_Q9_RTC72421', if_true: files('q9_rtc72421.c'))
